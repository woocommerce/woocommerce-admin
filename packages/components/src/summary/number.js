/**
 * External dependencies
 */
import { Button, Tooltip } from '@wordpress/components';
import { sprintf, __ } from '@wordpress/i18n';
import classnames from 'classnames';
import ChevronDownIcon from 'gridicons/dist/chevron-down';
import { isNil, noop } from 'lodash';
import PropTypes from 'prop-types';
import { createElement } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Link from '../link';
import { Text } from '../experimental';

/**
 * A component to show a value, label, and optionally a change percentage and children node. Can also act as a link to a specific report focus.
 *
 * @param {Object} props
 * @param {Node} props.children
 * @param {number} props.delta Change percentage. Float precision is rendered as given.
 * @param {string} props.href
 * @param {string} props.hrefType
 * @param {boolean} props.isOpen
 * @param {string} props.label
 * @param {Function} props.onToggle
 * @param {string} props.prevLabel
 * @param {number|string} props.prevValue
 * @param {boolean} props.reverseTrend
 * @param {boolean} props.selected
 * @param {number|string} props.value
 * @param {Function} props.onLinkClickCallback
 * @return {Object} -
 */
const SummaryNumber = ( {
	children,
	delta,
	href,
	hrefType,
	isOpen,
	label,
	onToggle,
	prevLabel,
	prevValue,
	reverseTrend,
	selected,
	value,
	onLinkClickCallback,
} ) => {
	const liClasses = classnames( 'woocommerce-summary__item-container', {
		'is-dropdown-button': onToggle,
		'is-dropdown-expanded': isOpen,
	} );
	const classes = classnames( 'woocommerce-summary__item', {
		'is-selected': selected,
		'is-good-trend': reverseTrend ? delta < 0 : delta > 0,
		'is-bad-trend': reverseTrend ? delta > 0 : delta < 0,
	} );

	let screenReaderLabel =
		delta > 0
			? sprintf(
					__( 'Up %f%% from %s', 'woocommerce-admin' ),
					delta,
					prevLabel
			  )
			: sprintf(
					__( 'Down %f%% from %s', 'woocommerce-admin' ),
					Math.abs( delta ),
					prevLabel
			  );
	if ( ! delta ) {
		screenReaderLabel = sprintf(
			__( 'No change from %s', 'woocommerce-admin' ),
			prevLabel
		);
	}

	let Container;
	const containerProps = {
		className: classes,
		'aria-current': selected ? 'page' : null,
	};

	if ( onToggle || href ) {
		const isButton = !! onToggle;
		Container = isButton ? Button : Link;
		if ( isButton ) {
			containerProps.onClick = onToggle;
			containerProps[ 'aria-expanded' ] = isOpen;
		} else {
			containerProps.href = href;
			containerProps.role = 'menuitem';
			containerProps.onClick = onLinkClickCallback;
			containerProps.type = hrefType;
		}
	} else {
		Container = 'div';
	}

	return (
		<li className={ liClasses }>
			<Container { ...containerProps }>
				<div className="woocommerce-summary__item-label">
					<Text variant="body.small" size="14" lineHeight="20px">
						{ label }
					</Text>
				</div>

				<div className="woocommerce-summary__item-data">
					<div className="woocommerce-summary__item-value">
						<Text variant="title.small" size="20" lineHeight="28px">
							{ ! isNil( value )
								? value
								: __( 'N/A', 'woocommerce-admin' ) }
						</Text>
					</div>

					<Tooltip
						text={
							! isNil( prevValue )
								? `${ prevLabel } ${ prevValue }`
								: __( 'N/A', 'woocommerce-admin' )
						}
						position="top center"
					>
						<div
							className="woocommerce-summary__item-delta"
							role="presentation"
							aria-label={ screenReaderLabel }
						>
							<Text variant="caption" size="12" lineHeight="16px">
								{ ! isNil( delta )
									? sprintf(
											__( '%f%%', 'woocommerce-admin' ),
											delta
									  )
									: __( 'N/A', 'woocommerce-admin' ) }
							</Text>
						</div>
					</Tooltip>
				</div>
				{ onToggle ? (
					<ChevronDownIcon
						className="woocommerce-summary__toggle"
						size={ 24 }
					/>
				) : null }
				{ children }
			</Container>
		</li>
	);
};

SummaryNumber.propTypes = {
	/**
	 * A number to represent the percentage change since the last comparison period - positive numbers will show
	 * a green up arrow, negative numbers will show a red down arrow, and zero will show a flat right arrow.
	 * If omitted, no change value will display.
	 */
	delta: PropTypes.number,
	/**
	 * An internal link to the report focused on this number.
	 */
	href: PropTypes.string,
	/**
	 * The type of the link
	 */
	hrefType: PropTypes.oneOf( [ 'wp-admin', 'wc-admin', 'external' ] )
		.isRequired,
	/**
	 * Boolean describing whether the menu list is open. Only applies in mobile view,
	 * and only applies to the toggle-able item (first in the list).
	 */
	isOpen: PropTypes.bool,
	/**
	 * A string description of this value, ex "Revenue", or "New Customers"
	 */
	label: PropTypes.string.isRequired,
	/**
	 * A function used to switch the given SummaryNumber to a button, and called on click.
	 */
	onToggle: PropTypes.func,
	/**
	 * A string description of the previous value's timeframe, ex "Previous year:".
	 */
	prevLabel: PropTypes.string,
	/**
	 * A string or number value to display - a string is allowed so we can accept currency formatting.
	 * If omitted, this section won't display.
	 */
	prevValue: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
	/**
	 * A boolean used to indicate that a negative delta is "good", and should be styled like a positive (and vice-versa).
	 */
	reverseTrend: PropTypes.bool,
	/**
	 * A boolean used to show a highlight style on this number.
	 */
	selected: PropTypes.bool,
	/**
	 * A string or number value to display - a string is allowed so we can accept currency formatting.
	 */
	value: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
	/**
	 * A function to be called after a SummaryNumber, rendered as a link, is clicked.
	 */
	onLinkClickCallback: PropTypes.func,
};

SummaryNumber.defaultProps = {
	href: '',
	hrefType: 'wc-admin',
	isOpen: false,
	prevLabel: __( 'Previous period:', 'woocommerce-admin' ),
	reverseTrend: false,
	selected: false,
	onLinkClickCallback: noop,
};

export default SummaryNumber;
