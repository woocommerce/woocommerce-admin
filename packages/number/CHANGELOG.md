# 2.2.0

-   Fix commonjs module build, allow package to be built in isolation. #7286

# 2.1.1

-   Update dependencies.
-   Add TypeScript support.

# 2.1.0

-   Update to @wordpress/eslint coding standards.

# 2.0.0

-   Remove lodash dependency.

## Breaking Changes

-   Decouple from global wcSettings object.
-   Exported methods of the number package have been rewritten to accept a configuration object as their first parameter.

# 1.0.4

-   Update dependencies.

# 1.0.3

-   Update license to GPL-3.0-or-later.

# 1.0.2

-   Bump dependency versions.

# 1.0.1

# 1.0.0

-   Initial release exports `numberFormat`, `formatValue`, and `calculateDelta`
