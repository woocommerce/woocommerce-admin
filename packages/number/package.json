{
	"name": "@woocommerce/number",
	"version": "2.2.0",
	"description": "Number formatting utilities for WooCommerce.",
	"author": "Automattic",
	"license": "GPL-3.0-or-later",
	"keywords": [
		"wordpress",
		"woocommerce"
	],
	"homepage": "https://github.com/woocommerce/woocommerce-admin/tree/main/packages/number/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/woocommerce/woocommerce-admin.git"
	},
	"bugs": {
		"url": "https://github.com/woocommerce/woocommerce-admin/issues"
	},
	"main": "build/index.js",
	"module": "build-module/index.js",
	"react-native": "src/index",
	"dependencies": {
		"locutus": "2.0.15"
	},
	"publishConfig": {
		"access": "public"
	},
	"scripts": {
		"clean": "pnpm exec rimraf tsconfig.tsbuildinfo build build-*",
		"build": "tsc --build ./tsconfig.json ./tsconfig-cjs.json",
		"start": "tsc --build --watch",
		"prepack": "pnpm run clean && pnpm run build",
		"test": "pnpm run build && pnpm run test:nobuild",
		"test:nobuild": "jest --config ./jest.config.json",
		"test-staged": "jest --bail --config ./jest.config.json --findRelatedTests"
	}
}
