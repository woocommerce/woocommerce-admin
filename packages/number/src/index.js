const numberFormatter = require( 'locutus/php/strings/number_format' );

/**
 * Number formatting configuration object
 *
 * @typedef {Object} NumberConfig
 * @property {number} [precision]         Decimal precision.
 * @property {string} [decimalSeparator]  Decimal separator.
 * @property {string} [thousandSeparator] Character used to separate thousands groups.
 */

/**
 * Formats a number using site's current locale
 *
 * @see http://locutus.io/php/strings/number_format/
 * @param {NumberConfig}  numberConfig Number formatting configuration object.
 * @param {number|string} number       number to format
 * @return {string} A formatted string.
 */
export function numberFormat(
	{ precision = null, decimalSeparator = '.', thousandSeparator = ',' },
	number
) {
	if ( typeof number !== 'number' ) {
		number = parseFloat( number );
	}

	if ( isNaN( number ) ) {
		return '';
	}

	let parsedPrecision = parseInt( precision, 10 );

	if ( isNaN( parsedPrecision ) ) {
		const [ , decimals ] = number.toString().split( '.' );
		parsedPrecision = decimals ? decimals.length : 0;
	}

	return numberFormatter(
		number,
		parsedPrecision,
		decimalSeparator,
		thousandSeparator
	);
}

/**
 * Formats a number as average or number string according to the given `type`.
 *  - `type = 'average'` returns a rounded `Number`
 *  - `type = 'number'` returns a formatted `String`
 *
 * @param {NumberConfig} numberConfig number formatting configuration object.
 * @param {string}       type         of number to format, `'average'` or `'number'`
 * @param {number}       value        to format.
 * @return {string | number | null} A formatted string.
 */
export function formatValue( numberConfig, type, value ) {
	if ( ! Number.isFinite( value ) ) {
		return null;
	}

	switch ( type ) {
		case 'average':
			return Math.round( value );
		case 'number':
			return numberFormat( { ...numberConfig, precision: null }, value );
	}
}

/**
 * Calculates the delta/percentage change between two numbers.
 *
 * @param {number} primaryValue   the value to calculate change for.
 * @param {number} secondaryValue the baseline which to calculdate the change against.
 * @return {?number} Percent change between the primaryValue from the secondaryValue.
 */
export function calculateDelta( primaryValue, secondaryValue ) {
	if (
		! Number.isFinite( primaryValue ) ||
		! Number.isFinite( secondaryValue )
	) {
		return null;
	}

	if ( secondaryValue === 0 ) {
		return 0;
	}

	return Math.round(
		( ( primaryValue - secondaryValue ) / secondaryValue ) * 100
	);
}
