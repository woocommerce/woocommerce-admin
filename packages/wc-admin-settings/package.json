{
	"name": "@woocommerce/wc-admin-settings",
	"version": "1.1.0",
	"description": "Wrapper around the wcSettings object - used for development.",
	"author": "Automattic",
	"license": "GPL-2.0-or-later",
	"homepage": "https://github.com/woocommerce/woocommerce-admin/tree/main/packages/wc-admin-settings/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/woocommerce/woocommerce-admin.git",
		"directory": "packages/wc-admin-settings"
	},
	"bugs": {
		"url": "https://github.com/woocommerce/woocommerce-admin/issues"
	},
	"files": [
		"index.js"
	],
	"main": "build/index.js",
	"module": "build-module/index.js",
	"private": true,
	"dependencies": {
		"@wordpress/i18n": "3.17.0"
	},
	"scripts": {
		"clean": "npx rimraf tsconfig.tsbuildinfo build build-*",
		"build": "tsc --build ./tsconfig.json ./tsconfig-cjs.json",
		"prepack": "npm run clean && npm run build",
		"start": "tsc --build --watch"
	}
}
