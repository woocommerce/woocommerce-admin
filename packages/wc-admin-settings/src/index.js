/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

// Remove mutable data from settings object to prevent access. Data stores should be used instead.
const mutableSources = [ 'wcAdminSettings', 'preloadSettings' ];
const settings = typeof wcSettings === 'object' ? wcSettings : {};
const SOURCE = Object.keys( settings ).reduce( ( source, key ) => {
	if ( ! mutableSources.includes( key ) ) {
		source[ key ] = settings[ key ];
	}
	return source;
}, {} );
Object.keys( settings.admin || {} ).forEach( ( key ) => {
	if ( ! mutableSources.includes( key ) ) {
		SOURCE[ key ] = settings.admin[ key ];
	}
} );

export const ADMIN_URL = SOURCE.adminUrl;
export const COUNTRIES = SOURCE.countries;
export const CURRENCY = SOURCE.currency;
export const LOCALE = SOURCE.locale;
export const ORDER_STATUSES = SOURCE.orderStatuses;
export const SITE_TITLE = SOURCE.siteTitle;
export const WC_ASSET_URL = SOURCE.wcAssetUrl;

/**
 * Retrieves a setting value from the setting state.
 *
 * @param {string}   name                         The identifier for the setting.
 * @param {*}    [fallback=false]             The value to use as a fallback
 *                                                if the setting is not in the
 *                                                state.
 * @param {Function} [filter=( val ) => val]  	  A callback for filtering the
 *                                                value before it's returned.
 *                                                Receives both the found value
 *                                                (if it exists for the key) and
 *                                                the provided fallback arg.
 *
 * @return {*}  The value present in the settings state for the given
 *                   name.
 */
export function getSetting( name, fallback = false, filter = ( val ) => val ) {
	if ( mutableSources.includes( name ) ) {
		throw new Error(
			__( 'Mutable settings should be accessed via data store.' )
		);
	}
	const value = SOURCE.hasOwnProperty( name ) ? SOURCE[ name ] : fallback;
	return filter( value, fallback );
}

/**
 * Sets a value to a property on the settings state.
 *
 * NOTE: This feature is to be removed in favour of data stores when a full migration
 * is complete.
 *
 * @deprecated
 *
 * @param {string}   name                        The setting property key for the
 *                                               setting being mutated.
 * @param {*}    value                       The value to set.
 * @param {Function} [filter=( val ) => val]     Allows for providing a callback
 *                                               to sanitize the setting (eg.
 *                                               ensure it's a number)
 */
export function setSetting( name, value, filter = ( val ) => val ) {
	if ( mutableSources.includes( name ) ) {
		throw new Error(
			__( 'Mutable settings should be mutated via data store.' )
		);
	}
	SOURCE[ name ] = filter( value );
}

/**
 * Returns a string with the site's wp-admin URL appended. JS version of `admin_url`.
 *
 * @param {string} path Relative path.
 * @return {string} Full admin URL.
 */
export function getAdminLink( path ) {
	return ( ADMIN_URL || '' ) + path;
}

/**
 * Adds a script to the page if it has not already been loaded. JS version of `wp_enqueue_script`.
 *
 * @param {Object} script WP_Script
 * @param {string} script.handle Script handle.
 * @param {string} script.src Script URL.
 */
export function enqueueScript( script ) {
	return new Promise( ( resolve, reject ) => {
		if ( document.querySelector( `#${ script.handle }-js` ) ) {
			resolve();
		}

		const domElement = document.createElement( 'script' );
		domElement.src = script.src;
		domElement.id = `${ script.handle }-js`;
		domElement.async = true;
		domElement.onload = resolve;
		domElement.onerror = reject;
		document.body.appendChild( domElement );
	} );
}
