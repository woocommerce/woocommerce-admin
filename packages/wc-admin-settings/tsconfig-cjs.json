{
    "extends": "../tsconfig-cjs",
    "compilerOptions": {
        "outDir": "build",
        "declaration": true
    }
}