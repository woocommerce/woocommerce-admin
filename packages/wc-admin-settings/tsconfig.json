{
	"extends": "../tsconfig",
	"compilerOptions": {
		"rootDir": "src",
		"outDir": "build-module",
		"declaration": true
	}
}