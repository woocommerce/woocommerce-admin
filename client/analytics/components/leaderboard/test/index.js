/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { numberFormat } from '@woocommerce/number';
import CurrencyFactory from '@woocommerce/currency';
import { CURRENCY } from '@woocommerce/wc-admin-settings';

/**
 * Internal dependencies
 */
import { Leaderboard } from '../';
import mockData from '../data/top-selling-products-mock-data';

const { formatAmount, formatDecimal } = CurrencyFactory( CURRENCY );

describe( 'Leaderboard', () => {
	test( 'should render empty message when there are no rows', () => {
		const { container } = render(
			<Leaderboard
				id="products"
				title={ '' }
				headers={ [] }
				rows={ [] }
				totalRows={ 5 }
			/>
		);

		expect( container ).toMatchSnapshot();
	} );

	test( 'should render correct data in the table', () => {
		const rows = mockData.map( ( row ) => {
			const {
				name,
				items_sold: itemsSold,
				net_revenue: netRevenue,
				orders_count: ordersCount,
			} = row;
			return [
				{
					display: '<a href="#">' + name + '</a>',
					value: name,
				},
				{
					display: numberFormat( CURRENCY, itemsSold ),
					value: itemsSold,
				},
				{
					display: numberFormat( CURRENCY, ordersCount ),
					value: ordersCount,
				},
				{
					display: formatAmount( netRevenue ),
					value: formatDecimal( netRevenue ),
				},
			];
		} );

		const headers = [
			{
				label: 'Name',
			},
			{
				label: 'Items sold',
			},
			{
				label: 'Orders',
			},
			{
				label: 'Net sales',
			},
		];
		const { container } = render(
			<Leaderboard
				id="products"
				title={ '' }
				headers={ headers }
				rows={ rows }
				totalRows={ 5 }
			/>
		);

		expect( container ).toMatchSnapshot();
	} );
} );
