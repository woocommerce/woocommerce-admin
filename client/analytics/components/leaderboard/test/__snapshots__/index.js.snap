// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard should render correct data in the table 1`] = `
<div>
  <div
    class="components-card is-size-medium woocommerce-table woocommerce-leaderboard css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-medium e1q7k77g1 css-jsa6v9-Flex-HeaderUI eboqfv50"
    >
      <h2
        class="css-1ahfdc3-Text e15wbhsk0"
        size="20"
      />
      <div
        class="woocommerce-table__actions"
      />
    </div>
    <div
      class="components-card__body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        aria-hidden="false"
        aria-labelledby="caption-0"
        class="woocommerce-table__table"
        role="group"
      >
        <table>
          <caption
            class="woocommerce-table__caption screen-reader-text"
            id="caption-0"
          >
            
          </caption>
          <tbody>
            <tr>
              <th
                class="woocommerce-table__header is-left-aligned"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Name
                </span>
              </th>
              <th
                class="woocommerce-table__header is-left-aligned"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Items sold
                </span>
              </th>
              <th
                class="woocommerce-table__header is-left-aligned"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Orders
                </span>
              </th>
              <th
                class="woocommerce-table__header is-left-aligned"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Net sales
                </span>
              </th>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item is-left-aligned"
                scope="row"
              >
                <div>
                  <a
                    href="#"
                  >
                    awesome shirt
                  </a>
                </div>
              </th>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  1,000.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  54.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  $999.99
                </div>
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item is-left-aligned"
                scope="row"
              >
                <div>
                  <a
                    href="#"
                  >
                    awesome pants
                  </a>
                </div>
              </th>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  90.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  41.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  $875.00
                </div>
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item is-left-aligned"
                scope="row"
              >
                <div>
                  <a
                    href="#"
                  >
                    awesome hat
                  </a>
                </div>
              </th>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  55.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  28.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  $75.75
                </div>
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item is-left-aligned"
                scope="row"
              >
                <div>
                  <a
                    href="#"
                  >
                    awesome sticker
                  </a>
                </div>
              </th>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  10.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  14.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  $24.50
                </div>
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item is-left-aligned"
                scope="row"
              >
                <div>
                  <a
                    href="#"
                  >
                    awesome button
                  </a>
                </div>
              </th>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  1.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  1.00
                </div>
              </td>
              <td
                class="woocommerce-table__item is-left-aligned"
              >
                <div>
                  $0.99
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="components-flex components-card__footer is-size-medium e1q7k77g4 css-5hda2c-Flex-FooterUI eboqfv50"
    />
  </div>
</div>
`;

exports[`Leaderboard should render empty message when there are no rows 1`] = `
<div>
  <div
    class="components-card is-size-medium woocommerce-leaderboard css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-medium e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      <h3
        class="css-1ahfdc3-Text e15wbhsk0"
        size="20"
      />
    </div>
    <div
      class="components-card__body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="woocommerce-table is-empty"
        style="--number-of-rows: 5;"
      >
        No data recorded for the selected time period.
      </div>
    </div>
  </div>
</div>
`;
